pytest_fxa/plugin.py

```diff
@@ -74,20 +89,25 @@ def fxa_account(fxa_client, fxa_email):
     password = ''.join([random.choice(string.ascii_letters) for i in range(8)])
     FxAccount = collections.namedtuple('FxAccount', 'email password')
     fxa_account = FxAccount(email=account.email, password=password)
-    session = fxa_client.create_account(fxa_account.email,
-                                        fxa_account.password)
-    logger.info('Created: {}'.format(fxa_account))
-    account.fetch()
-    message = account.wait_for_email(lambda m: 'x-verify-code' in m['headers'])
-    session.verify_email_code(message['headers']['x-verify-code'])
-    logger.info('Verified: {}'.format(fxa_account))
-    yield fxa_account
     try:
-        account.clear()
-        fxa_client.destroy_account(fxa_account.email, fxa_account.password)
-        logger.info('Removed: {}'.format(fxa_account))
-    except ClientError as e:
-        # 'Unknown Account' error is ok - account already deleted
-        # https://github.com/mozilla/fxa-auth-server/blob/master/docs/api.md#response-format
-        if e.errno not in [102]:
-            raise
+        session = fxa_client.create_account(fxa_account.email,
+                                            fxa_account.password)
+        logger.info('Created: {}'.format(fxa_account))
+        account.fetch()
+        message = account.wait_for_email(
+            lambda m: 'x-verify-code' in m['headers'] and
+            session.uid == m['headers']['x-uid']
+        )
+        session.verify_email_code(message['headers']['x-verify-code'])
+        logger.info('Verified: {}'.format(fxa_account))
+    except ClientError:
+        raise
+    except TypeError:
+        # Account didn't get verified, server didn't respond or returned `None`
+        raise RuntimeError("The account {} could not be verified.".format(
+            fxa_account.email)
+        )
+    else:
+        yield fxa_account
```

Comment:
The `yield` shouldn't need to be within an `else` block.

Reply:
Why is that? If I don't have a ```yield``` it seems it doesn't return the account. Using ```return``` doesn't work.

Reply:
I guess I am missing something. I don't see where else I would put it.

Reply:
You yield the account when it's successfully verified, so put it inside the `try` immediately after verification.

Reply:
It didn't seem to make sense 'trying' that but okay

Reply:
Why doesn't it make sense to include the yield within the try? Creating, verifying and ultimately yielding the account is exactly what this fixture is trying to do. If you want to yield to be outside of the `try` then you can probably also put it after the `finally` block.

Reply:
According to the python docs: ```The use of the else clause is better than adding additional code to the try clause because it avoids accidentally catching an exception that wasn’t raised by the code being protected by the try … except statement.```

Reply:
Sure, but yielding a variable would never raise an exception.